src/cider/nrepl/middleware/inspect.clj

```diff
@@ -47,77 +48,35 @@
   [handler msg]
   (handler (eval-msg msg)))
 
-(defn pop-reply
-  [{:keys [transport] :as msg}]
-  (try
-    (let [inspector (swap-inspector! msg inspect/up)]
-      (transport/send
-       transport
-       (response-for msg :value (:rendered inspector) :status :done)))
-    (catch Exception e
-      (transport/send
-       transport
-       (response-for msg (u/err-info e :inspect-pop-error))))))
-
-(defn push-reply
-  [{:keys [idx transport] :as msg}]
-  (try
-    (let [inspector (swap-inspector! msg inspect/down idx)]
-      (transport/send
-       transport
-       (response-for msg :value (:rendered inspector) :status :done)))
-    (catch Exception e
-      (transport/send
-       transport
-       (response-for msg (u/err-info e :inspect-push-error))))))
-
-(defn refresh-reply
-  [{:keys [transport] :as msg}]
-  (try
-    (let [inspector (swap-inspector! msg #(or % (inspect/fresh)))]
-      (transport/send
-       transport
-       (response-for msg :value (:rendered inspector) :status :done)))
-    (catch Exception e
-      (transport/send
-       transport
-       (response-for msg (u/err-info e :inspect-refresh-error))))))
-
-(defn next-page-reply
-  [{:keys [transport] :as msg}]
-  (try
-    (let [inspector (swap-inspector! msg inspect/next-page)]
-      (transport/send
-       transport
-       (response-for msg :value (:rendered inspector) :status :done)))
-    (catch Exception e
-      (transport/send
-       transport
-       (response-for msg (u/err-info e :inspect-next-page-error))))))
-
-(defn prev-page-reply
-  [{:keys [transport] :as msg}]
-  (try
-    (let [inspector (swap-inspector! msg inspect/prev-page)]
-      (transport/send
-       transport
-       (response-for msg :value (:rendered inspector) :status :done)))
-    (catch Exception e
-      (transport/send
-       transport
-       (response-for msg (u/err-info e :inspect-prev-page-error))))))
-
-(defn set-page-size-reply
-  [{:keys [page-size transport] :as msg}]
-  (try
-    (let [inspector (swap-inspector! msg inspect/set-page-size page-size)]
-      (transport/send
-       transport
-       (response-for msg :value (:rendered inspector) :status :done)))
-    (catch Exception e
-      (transport/send
-       transport
-       (response-for msg (u/err-info e :inspect-set-page-size-error))))))
+(defn- success [{:keys [transport] :as msg} inspector]
+  (transport/send transport (response-for msg :value (:rendered inspector) :status :done)))
+
+(defn- failure [{:keys [transport] :as msg} err err-kw]
+  (transport/send transport (base-error-response msg err err-kw :done)))
+
+(defn pop-reply [msg]
+  (try (success msg (swap-inspector! msg inspect/up))
+       (catch Exception e (failure msg e :inspect-pop-error))))
```

Comment:
I wonder if the Elisp code cares about those different error codes. Seems to me there's still room to cleanup/simplify all those ops.